x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1561,6 +1563,30 @@ static void validateForFips(Settings settings) {
             }
         });
 
+        Set<String> foundProviders = new HashSet<>();
+        for (Provider provider : java.security.Security.getProviders()) {
+            foundProviders.add(provider.getName().toLowerCase(Locale.ROOT));
+            if (logger.isTraceEnabled()) {
+                logger.trace("Security Provider: " + provider.getName() + ", Version: " + provider.getVersionStr());
```

Comment:
How is this tracing used in a debug scenario? It's very verbose. To find a typo in the config?

Reply:
It just echo's out the all the providers, versions, and supported ciphers, algorithms, etc. It is a bit divorced from what is used at runtime or what you config (e.g. `ssl.cipher_suites`). It can be handy to know which provider is providing with alogorithm. For example SunJCE provides Mac.HmacSHA512 so that provider (or an alternative) is required for `PBKDF2withHMACSHA512` which is heavily used for hashing. It can be hard to find this information from the various providers documentation.